index.js

```diff
@@ -125,7 +125,10 @@ Discovery.prototype.destroy = function (cb) {
     self.tracker.removeListener('peer', self._onTrackerPeer)
     self.tracker.removeListener('update', self._onTrackerAnnounce)
     tasks.push(function (cb) {
-      self.tracker.destroy(cb)
+      process.nextTick(function () {
+        self.tracker.destroy(cb)
```

Comment:
I just realized what you meant. This handles more then UDP.

Let me go back to the drawing board.